### PR TITLE
fix(wi-modal): fix top position for wi modal to close the modal

### PR DESCRIPTION
**Pull request by SMahil**
**WIP**
fix top position for wi modal to close the modal

Fixes https://openshift.io/openshiftio/openshiftio/plan/detail/1666

### CODE REVIEW

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2329-1 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2329-fabric8-planner.badger.fabric8.io

File: src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.less [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-
 .f8-wi-modal {
   position: absolute;
   top: 0;
   left: 0;
   right: 0;
   bottom: 0;
-  z-index: 3;
+  z-index: 9999;
   padding: (@grid-gutter-width/2);
```
Comment (nimishamukherjee) [inline src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.less]: @mindreeper2420 can you confirm the z-index value?

Comment (mindreeper2420) [reply src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.less]: All modals should be pre-determined by the frameworks. If something has to be set at this high of an amount, then there are other factors that need to be addressed.

The story that this fixes does not have a lot of information as to what the problem is.

**Comment by mindreeper2420** [comment]
@SMahil @nimishamukherjee This is most likely due to the fact that OpenShift.io now uses a fixed header. In order to fix this, instead of setting the z-index to be so high, the overall positioning should be moved from `top: 0;` to `top: 75px;`.

**Comment by SMahil** [comment]
@Adam is that ok to show header when page acts as modal... as in modals page beneath to it is not clickable

**Comment by SMahil** [comment]
@Adam ok...will submit the change with top position

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2329-2 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2329-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2329-4 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2329-fabric8-planner.badger.fabric8.io

File: src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.ts [hunk 2]
```diff
@@ -35,12 +36,33 @@ export class WorkItemDetailAddTypeSelectorWidgetComponent implements OnInit {
...
+  // ngDoCheck() {
+  //   if(this.modalPosition) {
+  //     let hdrHeight:number = 0;
+  //     if(document.getElementsByClassName('navbar-pf').length > 0) {
+  //       hdrHeight = (document.getElementsByClassName('navbar-pf')[0] as HTMLElement).offsetHeight;
+  //     }
+  //     console.log("HDRHEIGHT");
+  //     this.renderer.setStyle(this.modalPosition.nativeElement, 'top', hdrHeight + "px");
+  //   }
+  // }
```
Comment (vikram-raj) [inline src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.ts]: Remove these commented codes.

File: src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit, Input, OnChanges, EventEmitter, Output } from '@angular/core';
+import { Component, OnInit, DoCheck, Input, AfterViewChecked, OnChanges, EventEmitter, Output, ViewChild, ElementRef, Renderer2 } from '@angular/core';
```
Comment (vikram-raj) [inline src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.ts]: Remove DoCheck from import as you are not using this.

File: src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.ts [hunk 4]
```diff
@@ -21,11 +21,12 @@ import {
   templateUrl: './work-item-create-selector.component.html',
   styleUrls: ['./work-item-create-selector.component.less']
 })
-export class WorkItemDetailAddTypeSelectorWidgetComponent implements OnInit {
+export class WorkItemDetailAddTypeSelectorWidgetComponent implements OnInit , AfterViewChecked{
```
Comment (vikram-raj) [inline src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.ts]: Remove this extra space in between OnInit and ,

**Comment by SMahil** [comment]
@rgarg1 @jarifibrahim Functional tests are working fine locally. Can you please check this PR?

**Comment by jarifibrahim** [comment]
@SMahil The tests failures are intermittent and I haven't been to reproduce these failures locally. Me and @rgarg1 are investigating these failures.

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2329-17 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2329-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2329-18 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2329-fabric8-planner.badger.fabric8.io

File: src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit, Input, OnChanges, EventEmitter, Output } from '@angular/core';
+import { Component, OnInit, Input, AfterViewChecked, OnChanges, EventEmitter, Output, ViewChild, ElementRef, Renderer2 } from '@angular/core';
```
Comment (sanbornsen) [inline src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.ts]: It exceeds the standard number of columns. Can you break it into lines?

Comment (SMahil) [reply src/app/components/work-item-create/work-item-create-selector/work-item-create-selector.component.ts]: sure

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2329-19 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2329-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2329-20 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2329-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2329-22 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2329-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2329-23 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2329-fabric8-planner.badger.fabric8.io

**Pull request merged**